packages/fetchai/connections/soef/connection.py

```diff
@@ -251,9 +252,13 @@ def __init__(
         self.base_url = "http://{}:{}".format(soef_addr, soef_port)
         self.oef_search_dialogues = OefSearchDialogues()
 
+        data_dir = data_dir or os.getcwd()
```

Comment:
no no; there is no need to create this here. The framework will have to provide it!